packages/signers/src/fee-payer-signer.ts

```diff
@@ -16,14 +16,16 @@ export function setTransactionMessageFeePayerSigner<
     TTransactionMessage extends BaseTransactionMessage,
 >(
     feePayerSigner: TransactionSigner<TFeePayerAddress>,
-    transactionMessage: TTransactionMessage | (ITransactionMessageWithFeePayer<string> & TTransactionMessage),
-): ITransactionMessageWithFeePayerSigner<TFeePayerAddress> & TTransactionMessage {
+    transactionMessage: TTransactionMessage,
+): ITransactionMessageWithFeePayerSigner<TFeePayerAddress> & Omit<TTransactionMessage, 'feePayer' | 'feePayerSigner'> {
     if ('feePayer' in transactionMessage && feePayerSigner.address === transactionMessage.feePayer) {
         if ('feePayerSigner' in transactionMessage)
-            return transactionMessage as ITransactionMessageWithFeePayerSigner<TFeePayerAddress> & TTransactionMessage;
+            return transactionMessage as ITransactionMessageWithFeePayerSigner<TFeePayerAddress> &
+                Omit<TTransactionMessage, 'feePayer' | 'feePayerSigner'>;
```

Comment:
@lorisleiva, this code doesn't let someone change the `TransactionSigner`. If I have a message where the fee payer is Loris(A)/Phantom and I want to change it to Loris(A)/Solflare. This won't. Thoughts?

Reply:
Hmm that's a good point. We should totally override the signer with the one provided even when the addresses match as this is a totally valid use-case.

Reply:
#2897.